File: problems/2401 - 2800/2452/2452.py
class Solution:
    def twoEditWords(self, queries: List[str], dictionary: List[str]) -> List[str]:
        res: List[str] = []
        seen: dict[str, int] = {query: queries.count(query) for query in queries}
        for query in queries:
            for word in dictionary:
                cnt = 0
                if len(query) != len(word):
                    pass
                for i in range(len(word)):
                    if query[i] != word[i]:
                        cnt += 1
                if cnt <= 2 and seen[query]:
                    res.append(query)
                    seen[query] -= 1
                    break
        return res

File: problems/2401 - 2800/2452/2452.cpp
#include <vector>
#include <string>
#include <unordered_map>

class Solution {
public:
    std::vector <std::string> twoEditWords(std::vector <std::string>& queries, std::vector <std::string>& dictionary) {
        std::vector <std::string> res;
        std::unordered_map <std::string, int> seen;

        for (std::string& query1 : queries) {
            int cnt = 0;
            for (std::string& query2 : queries) {
                if (query1 == query2) {
                    ++cnt;
                }
            }
            seen[query1] = cnt;
        }

        for (std::string& query : queries) {
            for (std::string& word : dictionary) {
                int cnt = 0;
                if (query.size() != word.size()) {
                    continue;
                }
                for (int i = 0; i < word.size(); ++i) {
                    if (query[i] != word[i]) {
                        ++cnt;
                    }
                }
                if (cnt <= 2 && seen[query]) {
                    res.push_back(query);
                    --seen[query];
                    break;
                }
            }
        }
        return res;
    }
};
